core/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -2111,6 +2131,67 @@ describe("IModelTransformer", () => {
     }
   });
 
+  it("should transform 3d elements and their textures in target iModel", async () => {
+    // create source iModel
+    const sourceDbFile: string = IModelTransformerTestUtils.prepareOutputFile("IModelTransformer", "Transform3d-Source.bim");
+    const sourceDb = SnapshotDb.createEmpty(sourceDbFile, { rootSubject: { name: "Transform3d-Source" } });
+    const categoryId = SpatialCategory.insert(sourceDb, IModel.dictionaryId, "SpatialCategory", { color: ColorDef.green.toJSON() });
+    const category = sourceDb.elements.getElement<SpatialCategory>(categoryId);
+    const sourceModelId = PhysicalModel.insert(sourceDb, IModel.rootSubjectId, "Physical");
+    const renderMaterialId = RenderMaterialElement.insert(sourceDb, IModel.dictionaryId, "TextureMaterial", {
+      paletteName: "something",
+    });
+    const textureId = Texture.insertTexture(sourceDb, IModel.dictionaryId, "Texture", ImageSourceFormat.Png, BackendTestUtils.samplePngTexture.base64, "one white pixel");
+    const renderMaterial = sourceDb.elements.getElement<RenderMaterialElement>(renderMaterialId);
+    // update the texture id into the model so that they are processed out of order (material exported before texture)
+    if (renderMaterial.jsonProperties.materialAssets.renderMaterial.Map === undefined)
+      renderMaterial.jsonProperties.materialAssets.renderMaterial.Map = {};
+    if (renderMaterial.jsonProperties.materialAssets.renderMaterial.Map.Pattern === undefined)
+      renderMaterial.jsonProperties.materialAssets.renderMaterial.Map.Pattern = {};
+    if (renderMaterial.jsonProperties.materialAssets.renderMaterial.Map.Normal === undefined)
+      renderMaterial.jsonProperties.materialAssets.renderMaterial.Map.Normal = {};
+    renderMaterial.jsonProperties.materialAssets.renderMaterial.Map.TextureId = textureId;
```

Comment:
Ideally, test all possible combinations (normal map, pattern map, both).
At the least, test with normal texture Id != pattern texture Id.